test/base/src/androidTest/kotlin/io/realm/kotlin/test/shared/RealmTests.kt

```diff
@@ -494,6 +510,123 @@ class RealmTests {
         }
     }
 
+    private fun createWriteCopyLocalConfig(name: String, encryptionKey: ByteArray? = null): RealmConfiguration {
+        val builder = RealmConfiguration.Builder(schema = setOf(Parent::class, Child::class))
+            .directory(tmpDir)
+            .name(name)
+        if (encryptionKey != null) {
+            builder.encryptionKey(encryptionKey)
+        }
+        return builder.build()
+    }
+
+    @Test
+    fun writeCopyTo_localToLocal() {
+        val configA: RealmConfiguration = createWriteCopyLocalConfig("fileA.realm")
+        val configB: RealmConfiguration = createWriteCopyLocalConfig("fileB.realm")
+        Realm.open(configA).use { realm ->
+            realm.writeBlocking {
+                repeat(1000) { i: Int ->
+                    copyToRealm(
+                        Parent().apply {
+                            name = "Object-$i"
+                        }
+                    )
+                }
+            }
+            realm.writeCopyTo(configB)
+        }
+        // Copy is compacted i.e. smaller than original.
+        val fileASize: Long = platformFileSystem.metadata(configA.path.toPath()).size!!
+        val fileBSize: Long = platformFileSystem.metadata(configB.path.toPath()).size!!
+        assertTrue(fileASize >= fileBSize, "$fileASize >= $fileBSize")
+        // Content is copied
+        Realm.open(configB).use { realm ->
+            assertEquals(1000, realm.query<Parent>().count().find())
+        }
+    }
+
+    @Test
+    fun writeCopyTo_localToLocalEncrypted() {
+        val configA: RealmConfiguration = createWriteCopyLocalConfig("fileA.realm")
+        val configBEncrypted: RealmConfiguration = createWriteCopyLocalConfig("fileB.realm", Random.Default.nextBytes(Realm.ENCRYPTION_KEY_LENGTH))
+
+        // Write non-encrypted data
+        Realm.open(configA).use { realm ->
+            realm.writeBlocking {
+                repeat(1000) { i: Int ->
+                    copyToRealm(
+                        Parent().apply {
+                            name = "Object-$i"
+                        }
+                    )
+                }
+            }
+
+            // Copy to encrypted Realm
+            realm.writeCopyTo(configBEncrypted)
+        }
+
+        // Ensure that new Realm is encrypted
+        val configBUnencrypted: RealmConfiguration = RealmConfiguration.Builder(schema = setOf(Parent::class, Child::class))
+            .directory(tmpDir)
+            .name("fileB.realm")
+            .build()
+
+        assertFailsWith<IllegalArgumentException> {
+            Realm.open(configBUnencrypted)
+        }
+
+        Realm.open(configBEncrypted).use { realm ->
+            assertEquals(1000, realm.query<Parent>().count().find())
+        }
+    }
+
+    @Test
+    fun writeCopyTo_localEncryptedToLocal() {
```

Comment:
If we try to cover all cases then I guess we should theoretically also have local-encrypted to local-encrypted, but I don't know if it is worth it 🤷